ecommerce/admin.py

```diff
@@ -46,8 +48,31 @@ class ProductAdmin(VersionAdmin):
     """Admin for Product"""
 
     model = Product
-    search_fields = ["description", "price"]
-    list_display = ["id", "description", "price", "is_active"]
+    search_fields = (
+        "description",
+        "courseruns__title",
+        "programruns__program__title",
+        "courseruns__courseware_id",
+        "programruns__program__readable_id",
+    )
+    list_display = ("id", "content_object", "description", "price", "is_active")
+    list_filter = ("content_type", "is_active")
+
+    def change_view(self, request, object_id, form_url="", extra_context=None):
+        product = Product.objects.get(id=object_id)
+        extra_context = {"subtitle": self.content_object(product)}
+        return super(ProductAdmin, self).change_view(
+            request, object_id, form_url, extra_context
+        )
+
+    def content_object(self, obj):
+        """Return the content object details"""
+        if obj.content_type == ContentType.objects.get_for_model(CourseRun):
+            course_run = obj.purchasable_object
+            return f"{course_run.courseware_id} | {course_run.title}"
```

Comment:
The [default str formatting of a course run](https://github.com/mitodl/mitxonline/blob/main/courses/models.py#L543-L545) already supports this format and i think we can reuse that here. e.g. `str(course_run)`.

As for the program run on line#75 we also have [str override](https://github.com/mitodl/mitxonline/blob/main/courses/models.py#L211-L212) for that but in that we. are using. run_tag as. the second parsm, i think. we can standardize that there and the use the str format of ProgramRun here as well.

Reply:
Awesome suggestion. Had a look at the product model's __str__ but that wasn't a feasible option. This one works great.